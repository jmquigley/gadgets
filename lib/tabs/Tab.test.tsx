'use strict';

import {mount, shallow} from 'enzyme';
import * as React from 'react';
import {getDefaultTabProps, Tab} from './index';

test('Test retrieval of Tab props object', () => {
	const props = getDefaultTabProps();

	expect(props).toBeDefined();
	expect(props).toMatchSnapshot();
});

test('Test the creation of a Tab instance', () => {
	const ctl = shallow(
		<Tab title="tab title" selected>Test content</Tab>
	);

	expect(ctl).toBeDefined();
	expect(ctl).toMatchSnapshot();
});

test('Test disabling of a Tab', () => {
	const click = jest.fn();
	const ctl = mount(
		<Tab
			disabled={true}
			href={{
				selectHandler: click
			}}
			title="tab title"
		>
			Test content
		</Tab>
	);

	expect(ctl).toBeDefined();
	expect(ctl).toMatchSnapshot();

	ctl.find('.ui-label').first().simulate('click');
	expect(click).not.toHaveBeenCalled();
});

test('Test making a Tab invisible', () => {
	const ctl = mount(
		<Tab
			title="tab title"
			visible={false}
		>
		Test content
		</Tab>
	);

	expect(ctl).toBeDefined();
	expect(ctl).toMatchSnapshot();
});

test('Test the click handler in the Tab instance', () => {
	const click = jest.fn();
	const ctl = mount(
		<Tab
			href={{
				selectHandler: click
			}}
			title="tab title"
		>
			Test content
		</Tab>
	);

	expect(ctl).toBeDefined();
	expect(ctl).toMatchSnapshot();

	ctl.find('.ui-label').first().simulate('click');
	expect(click).toHaveBeenCalled();
});

test('Test the close click handler on a Tab instance', () => {
	const close = jest.fn();
	const hiddenHandler = jest.fn();
	const ctl = mount(
		<Tab
			href={{
				hiddenTabHandler: hiddenHandler
			}}
			onClose={close}
			title="tab title"
		>
			Test content
		</Tab>
	);

	expect(ctl).toBeDefined();
	expect(ctl).toMatchSnapshot();

	expect(ctl.state('hidden')).toBe(false);
	ctl.find('.ui-button').first().simulate('click');

	expect(hiddenHandler).toHaveBeenCalled();
	expect(hiddenHandler).toHaveBeenCalledWith(ctl.instance());

	expect(close).toHaveBeenCalled();
	expect(close).toHaveBeenCalledWith(ctl.instance());

	expect(ctl.state('hidden')).toBe(true);
});
