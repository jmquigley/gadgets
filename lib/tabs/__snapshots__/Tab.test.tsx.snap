// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test disabling of a Tab 1`] = `
<Tab
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  controlled={true}
  direction="right"
  disabled={true}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={false}
  href={
    Object {
      "selectHandler": [MockFunction],
    }
  }
  id={null}
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth="120px"
  noborder={false}
  noclose={false}
  noedit={false}
  nohover={false}
  nopropagation={false}
  notheme={false}
  notooltip={false}
  obj="Tab"
  onClick={[Function]}
  onClose={[Function]}
  orientation="top"
  padding={null}
  ripple={false}
  selected={false}
  sizing="normal"
  style={Object {}}
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  title="tab title"
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Test making a Tab invisible 1`] = `
<Tab
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={false}
  href={
    Object {
      "selectHandler": [Function],
    }
  }
  id={null}
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth="120px"
  noborder={false}
  noclose={false}
  noedit={false}
  nohover={false}
  nopropagation={false}
  notheme={false}
  notooltip={false}
  obj="Tab"
  onClick={[Function]}
  onClose={[Function]}
  orientation="top"
  padding={null}
  ripple={false}
  selected={false}
  sizing="normal"
  style={Object {}}
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  title="tab title"
  tooltip={null}
  top={null}
  visible={false}
  width={null}
  xcss={null}
/>
`;

exports[`Test retrieval of Tab props object 1`] = `
Object {
  "backgroundColor": null,
  "border": null,
  "borderColor": null,
  "bottom": null,
  "children": null,
  "className": null,
  "color": null,
  "controlled": true,
  "direction": "right",
  "disabled": false,
  "err": null,
  "errorMessage": null,
  "focus": false,
  "height": null,
  "hidden": false,
  "href": Object {
    "selectHandler": [Function],
  },
  "id": null,
  "left": null,
  "location": "none",
  "maxHeight": null,
  "maxWidth": null,
  "minHeight": null,
  "minWidth": "120px",
  "noborder": false,
  "noclose": false,
  "noedit": false,
  "nohover": false,
  "nopropagation": false,
  "notheme": false,
  "notooltip": false,
  "obj": "Tab",
  "onClick": [Function],
  "onClose": [Function],
  "orientation": "top",
  "padding": null,
  "ripple": false,
  "selected": false,
  "sizing": "normal",
  "style": Object {},
  "testing": true,
  "theme": Object {
    "backgroundColor": "#ffffff",
    "borderColor": "#c0c0c0",
    "buttonColor": "#666666",
    "chevronColor": "#cccccc",
    "color": "#000000",
    "fontFamily": "OpenSans, Georgia, serif",
    "headerBackgroundColor": "#004358",
    "headerForegroundColor": "#ffffff",
    "headerHoverColor": "#ffe11a",
    "hoverColor": "#c0c0c0",
    "inputBorderColor": "#c0c0c0",
    "itemHoverColor": "#bedb39",
    "outlineColor": "#51cbee",
    "searchFocus": "#51cbee",
    "searchMatch": "#fd7400",
    "selectedBackgroundColor": "#1f8a70",
    "selectedForegroundColor": "#ffe11a",
    "titleBarBackgroundColor": "#004358",
    "titleBarForegroundColor": "#ffe11a",
    "tooltipBackgroundColor": "#708090",
    "tooltipForegroundColor": "#ffffff",
    "transitionDelay": "0.33s",
  },
  "title": "",
  "tooltip": null,
  "top": null,
  "visible": true,
  "width": null,
  "xcss": null,
}
`;

exports[`Test retrieval of Tab props object 2`] = `
Object {
  "children": null,
  "sizing": "normal",
  "style": Object {},
  "visible": true,
}
`;

exports[`Test the click handler in the Tab instance 1`] = `
<Tab
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={false}
  href={
    Object {
      "selectHandler": [MockFunction],
    }
  }
  id={null}
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth="120px"
  noborder={false}
  noclose={false}
  noedit={false}
  nohover={false}
  nopropagation={false}
  notheme={false}
  notooltip={false}
  obj="Tab"
  onClick={[Function]}
  onClose={[Function]}
  orientation="top"
  padding={null}
  ripple={false}
  selected={false}
  sizing="normal"
  style={Object {}}
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  title="tab title"
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Test the close click handler on a Tab instance 1`] = `
<Tab
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={false}
  href={
    Object {
      "hiddenTabHandler": [MockFunction],
    }
  }
  id={null}
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth="120px"
  noborder={false}
  noclose={false}
  noedit={false}
  nohover={false}
  nopropagation={false}
  notheme={false}
  notooltip={false}
  obj="Tab"
  onClick={[Function]}
  onClose={[MockFunction]}
  orientation="top"
  padding={null}
  ripple={false}
  selected={false}
  sizing="normal"
  style={Object {}}
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  title="tab title"
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Test the creation of a Tab instance 1`] = `
<Wrapper
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={false}
  href={
    Object {
      "selectHandler": [Function],
    }
  }
  id={null}
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth="120px"
  noborder={false}
  noclose={false}
  noedit={false}
  nohover={false}
  nopropagation={false}
  notheme={false}
  notooltip={false}
  obj="Tab"
  onClick={[Function]}
  onClose={[Function]}
  onError={[Function]}
  orientation="top"
  padding={null}
  reset={false}
  ripple={false}
  selected={true}
  sizing="normal"
  style={Object {}}
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  title="tab title"
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
>
  <Tab__TabView
    className="ui-tab ui-selected"
    disabled={false}
    minWidth="120px"
    orientation="top"
    selected={true}
    sizing="normal"
    style={Object {}}
    visible={true}
    width={null}
  >
    <Item
      backgroundColor={null}
      border={null}
      borderColor={null}
      bottom={null}
      className={null}
      color={null}
      controlled={true}
      direction="right"
      disabled={false}
      err={null}
      errorMessage={null}
      focus={false}
      height={null}
      hidden={false}
      hiddenLeftButton={false}
      hiddenRightButton={true}
      href={
        Object {
          "selectHandler": [Function],
        }
      }
      id={null}
      layout={5}
      left={null}
      leftButton={null}
      location="none"
      maxHeight={null}
      maxWidth={null}
      minHeight={null}
      minWidth="120px"
      noborder={false}
      noclose={false}
      noedit={true}
      nohover={false}
      nopropagation={false}
      notheme={false}
      notooltip={false}
      obj="Tab"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onClose={[Function]}
      onDoubleClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onMouseOut={[Function]}
      onUpdate={[Function]}
      orientation="top"
      padding={null}
      rightButton={
        <Button
          backgroundColor={null}
          border={null}
          borderColor={null}
          bottom={null}
          className={null}
          color={null}
          controlled={true}
          direction="right"
          disabled={false}
          err={null}
          errorMessage={null}
          focus={false}
          height={null}
          hidden={false}
          href={
            Object {
              "selectHandler": [Function],
            }
          }
          iconName="times"
          iconStyle=""
          id={null}
          kbActivate=""
          left={null}
          location="none"
          maxHeight={null}
          maxWidth={null}
          minHeight={null}
          minWidth="120px"
          noborder={false}
          noclose={false}
          noedit={false}
          nohover={false}
          nopropagation={false}
          notheme={false}
          notooltip={false}
          obj="Tab"
          onClick={[Function]}
          onClose={[Function]}
          orientation="top"
          padding={null}
          ripple={false}
          selected={true}
          sizing="normal"
          style={Object {}}
          testing={true}
          theme={
            Object {
              "backgroundColor": "#ffffff",
              "borderColor": "#c0c0c0",
              "buttonColor": "#666666",
              "chevronColor": "#cccccc",
              "color": "#000000",
              "fontFamily": "OpenSans, Georgia, serif",
              "headerBackgroundColor": "#004358",
              "headerForegroundColor": "#ffffff",
              "headerHoverColor": "#ffe11a",
              "hoverColor": "#c0c0c0",
              "inputBorderColor": "#c0c0c0",
              "itemHoverColor": "#bedb39",
              "outlineColor": "#51cbee",
              "searchFocus": "#51cbee",
              "searchMatch": "#fd7400",
              "selectedBackgroundColor": "#1f8a70",
              "selectedForegroundColor": "#ffe11a",
              "titleBarBackgroundColor": "#004358",
              "titleBarForegroundColor": "#ffe11a",
              "tooltipBackgroundColor": "#708090",
              "tooltipForegroundColor": "#ffffff",
              "transitionDelay": "0.33s",
            }
          }
          title="tab title"
          tooltip={null}
          top={null}
          visible={true}
          width={null}
          xcss={null}
        >
          Test content
        </Button>
      }
      ripple={false}
      selected={true}
      sizing="normal"
      stacked={false}
      style={Object {}}
      testing={true}
      theme={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#c0c0c0",
          "buttonColor": "#666666",
          "chevronColor": "#cccccc",
          "color": "#000000",
          "fontFamily": "OpenSans, Georgia, serif",
          "headerBackgroundColor": "#004358",
          "headerForegroundColor": "#ffffff",
          "headerHoverColor": "#ffe11a",
          "hoverColor": "#c0c0c0",
          "inputBorderColor": "#c0c0c0",
          "itemHoverColor": "#bedb39",
          "outlineColor": "#51cbee",
          "searchFocus": "#51cbee",
          "searchMatch": "#fd7400",
          "selectedBackgroundColor": "#1f8a70",
          "selectedForegroundColor": "#ffe11a",
          "titleBarBackgroundColor": "#004358",
          "titleBarForegroundColor": "#ffe11a",
          "tooltipBackgroundColor": "#708090",
          "tooltipForegroundColor": "#ffffff",
          "transitionDelay": "0.33s",
        }
      }
      title="tab title"
      tooltip={null}
      top={null}
      useedit={false}
      visible={true}
      widget={null}
      width={null}
      xcss={null}
    >
      Test content
    </Item>
  </Tab__TabView>
</Wrapper>
`;
